Feature/4

dipteewarudkar opened the pull request.

duergner committed:
Merge branch 'hotfix/r0.3.3'

duergner committed:
Merge branch 'release/r0.4.0'

duergner committed:
Merge branch 'release/r0.5.0'

dipteewarudkar committed:
Adding model classes for Address
Adding service class for Address

TODO common Abstract Class

dipteewarudkar committed:
Fixing GetAddressesResponse structure

dipteewarudkar committed:
Fixing build warnings

dipteewarudkar committed:
Fixing update address to http PUT request

dipteewarudkar committed:
Adding wrapper for zalando Api for
Cart
checkout
order
customer

dipteewarudkar committed:
Fixing :
Formatting problems
Naming issues
Removing repeated objects

TODO AbstractService/postContructs -> Will keep this in hold till we
have a running version to test these APIs.

dipteewarudkar committed:
Formatting fixes

dipteewarudkar committed:
Changing response of read(customer_no) in addressService to
List<Address>

dipteewarudkar committed:
Fixing build javadoc errors

dipteewarudkar committed:
Fixing some more javadoc build error

dipteewarudkar committed:
1)Changing Currency to enum
2)Fixing naming bug.

dipteewarudkar commented:
@duergner For merging

dipteewarudkar committed:
Adding service for getting access token and some other small changes

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Delivery.java:
```diff
@@ -0,0 +1,34 @@
+public class Delivery extends Base {
+	
+	@JsonProperty
+	private String earliest;
```
We should maybe use `Date` here instead of `String` and apply correct `@JsonFormat`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Delivery.java:
```diff
@@ -0,0 +1,34 @@
+	private String earliest;
+	
+	@JsonProperty
+	private String latest;
```
We should maybe use `Date` here instead of `String` and apply correct `@JsonFormat`

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Order.java:
```diff
@@ -0,0 +1,98 @@
+	private Money taxTotal;
+
+	@JsonProperty
+	private String created;
```
We should maybe use `Date` here instead of `String` and apply correct `@JsonFormat`

dipteewarudkar committed:
Some changes to removing CreateOrderRequest and using the common Order
class.
Adding token property on Payment
Some Camel case fixes

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/model/Address.java:
```diff
@@ -0,0 +1,120 @@
+
+	public Address(){}
+	
+	public Address(String id, String customer_number, String gender, String first_name, String last_name, String street,String additional, String zip, String city, String country_code, Boolean pack_station,Boolean default_billing, Boolean default_shipping) {
```
Use camel case for parameters please

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -0,0 +1,15 @@
+
+	@Override
+	public String read() {
+		// TODO Auto-generated method stub
```
Is this empty by intention?

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AbstractService.java:
```diff
@@ -0,0 +1,285 @@
+package io.sprucehill.zalando.api.nativecart.service;
```
Can you use the already existing AbstractService now?

dipteewarudkar committed:
Using Abstract Service from internal package and handling Exceptions

dipteewarudkar committed:
Some changes needed while integrating with live zalando Apis

dipteewarudkar committed:
Changing StringEntity to ByteEntity to save special characters such as 
(ü) when posting data

Adding "problem" class to get details of the exceptions

Changing the api path for adding updating cart

Passing accessToken to get customer details

Adding clientId and clientCredentials in abstractservice

dipteewarudkar committed:
Fixing javadoc build errors

dipteewarudkar committed:
Fixing some more javadoc build errors

dipteewarudkar committed:
Changing StringEntity to ByteArrayEntity

dipteewarudkar committed:
Implemented Review  comments
1) Changed magic numbers to HttpHeaders
2)Added authorization on checkout and order services

dipteewarudkar commented:
@duergner Please recheck .Thanks!

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -17,15 +24,33 @@
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
+		
+		List<NameValuePair> params = new ArrayList<NameValuePair>();
+		params.add(new BasicNameValuePair("realm","%2Fcustomers"));
```
You can use `/` here; it will be correctly encoded on the entity

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -17,15 +24,33 @@
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
+		
+		List<NameValuePair> params = new ArrayList<NameValuePair>();
+		params.add(new BasicNameValuePair("realm","%2Fcustomers"));
```
You can use `/` here; it will be correctly encoded on the entity

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -17,15 +24,33 @@
-		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientCredential);
-		request.setHeader("Content-Type","application/x-www-form-urlencoded");
+		HttpPost request = postRequest("/access_token");
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
```
You might not need this here when using `UrlEncodedFormEntity` - this entity class should add the correct `Content-Type` header itself

duergner commented on src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java:
```diff
@@ -17,15 +24,33 @@
 	public TokenInfo readFromAuthorizationCode(String authCode,String redirectUrl) throws Exception {
 		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=authorization_code&client_id="+clientId+"&client_secret="+clientCredential+"&code="+authCode+"&redirect_uri="+redirectUrl);
-		request.setHeader("Content-Type","application/x-www-form-urlencoded");
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
```
You might not need this here when using `UrlEncodedFormEntity` - this entity class should add the correct `Content-Type` header itself

dipteewarudkar committed:
Implementing review comments , removing content-type setting 
and adding /customers instead of encoded value

dipteewarudkar commented:
@duergner One more time review plz  :)

duergner commented:
Looks fine - I'll merge it in later

dipteewarudkar committed:
Some bug fix came up while integration.
- Changing app_domain_id to sales channel.
- Linking corrrect property to shipping address id in checkout object 
- Updating the api url for checkout and order

dipteewarudkar committed:
Removing /customers/customer_number/ from urls

duergner committed:
Should bump the version to correct snapshot one
